assembl/views/api2/content_source.py

```diff
@@ -55,6 +61,9 @@ def fetch_posts(request):
             raise HTTPUnauthorized("Only discussion administrator\
                                    can "+'and'.join(requested))
 
+    if force_restart:
```

Comment:
What you're proposing makes a lot of sense to me. No objection at all to moving it downstream.